openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py

```diff
@@ -605,6 +605,45 @@ def fill_account_journal_payment_credit_debit_account_id(env):
         )
 
 
+def create_new_counterpar_account_payment_transfer(env):
+    # Create new counterpart payment with account payment transfer
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_payment (move_id, is_internal_transfer, partner_type,
+            payment_type,
+            amount, currency_id,
+            destination_account_id, partner_id, journal_id,
+            create_uid, create_date, write_uid, write_date)
+        SELECT move.id, true, ap.partner_type,
+            CASE
+            WHEN journal.id = ap.destination_journal_id THEN 'inbound' ELSE 'outbound'
+            END,
+            ap.amount, ap.currency_id,
+            ap.destination_account_id, ap.partner_id, move.journal_id,
+            ap.create_uid, ap.create_date, ap.write_uid, ap.write_date
+        FROM account_payment ap
+        JOIN account_move move
+            ON (move.payment_id = ap.id AND move.id != ap.move_id)
+        JOIN account_journal journal ON journal.id = move.journal_id
+        WHERE ap.payment_type = 'transfer'
+        """,
+    )
+
+
+def map_account_payment_transfer(env):
+    # map payment_type from transfer to 'inbound'/'outbound'
+    # and set is_internal_transfer as true on account payment transfer
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_payment ap
+        SET is_internal_transfer = true, payment_type = 'outbound'
+        WHERE ap.payment_type = 'transfer'
```

Comment:
I think here needs an additional condition, right? 🤔

Reply:
No, simply applying for the send transfer
For receiving transfer, it is created above

Reply:
Yeah, but the created above also will be changed with this query, that's what I mean.

Reply:
I mean, your are putting all transfers (old and new ones) to outbound, and that's what doesn't look correct to me.

Reply:
No, when created, it is already payment_type as outbound / inbound, not transfer

Reply:
Oh, right, my bad.